infer/inference.cc

```diff
@@ -300,8 +300,6 @@ unique_ptr<cfg::CFG> Inference::run(core::Context ctx, unique_ptr<cfg::CFG> cfg)
                         DEBUG_ONLY(histogramInc("untyped.sources", bind.bind.type.untypedBlame().rawId()););
                         if (auto e = ctx.beginError(bind.loc, core::errors::Infer::UntypedValue)) {
                             e.setHeader("This code is untyped");
-                            e.addErrorNote("Support for `{}` is minimal. Consider using `{}` instead.", "typed: strong",
-                                           "typed: strict");
```

Comment:
We should not delete this in the final feature.

One way to do this better might be:

1.  Introduce a new error class, like `core::errors::Infer::UntypedValueInformation`
2.  Make that error report at `StrictLevel::True`, instead of `StrictLevel::Strong`
3.  Build a helper function like `ErrorClass errorClassForUntyped(const GlobalState &gs, FileRef file)`, which takes on the new `isOpenInClient()` logic that currently resides in `shouldReportErrorOn`.
4.  Call that helper function as the argument to `beginError`:

    ```cpp
    auto what = errorClassForUntyped(gs, ctx.file)
    ctx.beginError(bind.loc, what)
    ```

5. Hide this call to `addErorrNote` behind a check like `if (what == core::errors::Infer::UntypedValue)`

This note is important because some Sorbet-internal features are powered by `T.untyped`. For example, it's not possible to use `# typed: strong` on a file that uses `super`, because Sorbet models `super` with `T.untyped`. So it's important that we expose this note when people _are_ trying to use `

Reply:
ack, patched!